compiler/z/env/OMRCPU.cpp

```diff
@@ -382,3 +382,91 @@ OMR::Z::CPU::isTargetWithinBranchRelativeRILRange(intptrj_t targetAddress, intpt
             (targetAddress % 2 == 0);
    }
+
+void
+OMR::Z::CPU::postInitialization()
+   {
+   // The following nested if statements cascade so as to have the effect of only enabling the least common denominator
+   // of disable CPU architectures. For example if the user specified to disable z13 when running on a z15 machine the
+   // logic below will ensure we only reach the `setProcessor` call for zEC12.
+   if (self()->isAtLeast(OMR_PROCESSOR_S390_Z10) &&
+         !TR::Options::getCmdLineOptions()->getOption(TR_DisableZ10))
+      {
+      self()->setProcessor(OMR_PROCESSOR_S390_Z10);
+
+      if (self()->isAtLeast(OMR_PROCESSOR_S390_Z196) &&
+            !TR::Options::getCmdLineOptions()->getOption(TR_DisableZ196))
+         {
+         self()->setProcessor(OMR_PROCESSOR_S390_Z196);
+
+         if (self()->isAtLeast(OMR_PROCESSOR_S390_ZEC12) &&
+               !TR::Options::getCmdLineOptions()->getOption(TR_DisableZEC12))
+            {
+            self()->setProcessor(OMR_PROCESSOR_S390_ZEC12);
+
+            if (self()->isAtLeast(OMR_PROCESSOR_S390_Z13) &&
+                  !TR::Options::getCmdLineOptions()->getOption(TR_DisableZ13))
+               {
+               self()->setProcessor(OMR_PROCESSOR_S390_Z13);
+
+               if (self()->isAtLeast(OMR_PROCESSOR_S390_Z14) &&
+                     !TR::Options::getCmdLineOptions()->getOption(TR_DisableZ14))
+                  {
+                  self()->setProcessor(OMR_PROCESSOR_S390_Z14);
+
+                  if (self()->isAtLeast(OMR_PROCESSOR_S390_Z15) &&
+                        !TR::Options::getCmdLineOptions()->getOption(TR_DisableZ15))
+                     {
+                     self()->getProcessor(OMR_PROCESSOR_S390_Z15);
+
+                     if (self()->isAtLeast(OMR_PROCESSOR_S390_ZNEXT) &&
+                           !TR::Options::getCmdLineOptions()->getOption(TR_DisableZNext))
+                        {
+                        self()->setProcessor(OMR_PROCESSOR_S390_ZNEXT);
+                        }
+                     }
+                  }
+               }
+            }
+         }
+      }
+
+   if (!self()->isAtLeast(OMR_PROCESSOR_S390_Z10))
+      {
+      self()->disableFeature(OMR_FEATURE_S390_DFP);
+      }
+
+   if (!self()->isAtLeast(OMR_PROCESSOR_S390_Z196))
+      {
+      self()->disableFeature(OMR_FEATURE_S390_HIGH_WORD);
+      }
+
+   if (!self()->isAtLeast(OMR_PROCESSOR_S390_ZEC12))
+      {
+      self()->disableFeature(OMR_FEATURE_S390_TE);
+      }
+
+   if (!self()->isAtLeast(OMR_PROCESSOR_S390_Z13))
+      {
+      self()->disableFeature(OMR_FEATURE_S390_VECTOR_FACILITY);
+      }
+
+   if (!self()->isAtLeast(OMR_PROCESSOR_S390_Z14))
+      {
+      self()->disableFeature(OMR_FEATURE_S390_VECTOR_PACKED_DECIMAL);
+
+      // turn off GS facility if GC has -XXgc:softwareRangeCheckReadBarrier
+      if (TR::Compiler->javaVM->memoryManagerFunctions->j9gc_software_read_barrier_enabled(TR::Compiler->javaVM))
```

Comment:
This condition doesn't make sense in OMR and either needs to be removed or rewritten to avoid being Java-specific.